Python/Include/Sample.hpp

```diff
@@ -59,6 +64,50 @@ namespace Details {
       std::make_index_sequence<std::tuple_size_v<typename S::Arguments>>()) };
     return result;
   }
+
+  class SampleConverter : private Noncopyable {
```

Comment:
This is going to be a problem on Windows. You will need to look into DLLEXPORT to ensure that different libraries linking to rover share the same singleton.

Reply:
Do you mean that a user can create their own Python export library (e.g. to export `Sample<float, float>`) while linking against Rover.pyd, which exports `PythonSample`? This is going to be tricky.

Reply:
Actually, `SampleConverter` is not even a template. Isn't just moving the definition of `SampleConverter::get_instance` to `Sample.cpp` enough?

Reply:
For this case feel free to leave it out, but add a case on Fogbugz about this issue so we can visit it later.

Reply:
https://eidolonsystems.fogbugz.com/f/cases/1459/Mixed-use-of-SampleConverter-Python-Sample-exports